pkgs/by-name/li/limine/package.nix

```
@@ -1,67 +1,115 @@
+  lib,
+  llvmPackages_18,
+  mtools,
+  nasm,
+  # The following options map to configure flags.
+  enableAll ? false,
+  buildCDs ? false,
+  targets ? [ ],
+  # x86 specific flags
+  biosSupport ? false,
+  pxeSupport ? false,
 }:
 
 let
-  version = "7.9.1";
+  stdenv = clang18Stdenv;
+  llvmPackages = llvmPackages_18;
+
+  version = "8.0.11";
+
+  hasI686 =
+    (if targets == [ ] then stdenv.hostPlatform.isx86_32 else (builtins.elem "i686" targets))
+    || enableAll;
+
+  hasX86_64 =
+    (if targets == [ ] then stdenv.hostPlatform.isx86_64 else (builtins.elem "x86_64" targets))
+    || enableAll;
+
+  uefiFlags =
+    target:
+    {
+      aarch64 = [ "--enable-uefi-aarch64" ];
+      i686 = [ "--enable-uefi-ia32" ];
+      loongarch64 = [ "--enable-uefi-loongarch64" ];
+      riscv64 = [ "--enable-uefi-riscv64" ];
+      x86_64 = [ "--enable-uefi-x86-64" ];
+    }
+    .${target} or (throw "Unsupported target ${target}");
+
+  configureFlags =
+    lib.optionals enableAll [ "--enable-all" ]
+    ++ lib.optionals biosSupport [ "--enable-bios" ]
+    ++ lib.optionals (buildCDs && biosSupport) [ "--enable-bios-cd" ]
+    ++ lib.optionals buildCDs [ "--enable-uefi-cd" ]
+    ++ lib.optionals pxeSupport [ "--enable-bios-pxe" ]
+    ++ lib.concatMap uefiFlags (
+      if targets == [ ] then [ stdenv.hostPlatform.parsed.cpu.name ] else targets
+    );
 in
+
+assert lib.assertMsg (!(biosSupport && !hasI686)) "BIOS builds are possible only for x86";
+
+assert lib.assertMsg (!(pxeSupport && !hasI686)) "PXE builds are possible only for x86";
+
 # The output of the derivation is a tool to create bootable images using Limine
 # as bootloader for various platforms and corresponding binary and helper files.
 stdenv.mkDerivation {
-  inherit version;
+  inherit version configureFlags;
   pname = "limine";
   # We don't use the Git source but the release tarball, as the source has a
   # `./bootstrap` script performing network access to download resources.
   # Packaging that in Nix is very cumbersome.
   src = fetchurl {
     url = "https://github.com/limine-bootloader/limine/releases/download/v${version}/limine-${version}.tar.gz";
-    hash = "sha256-cR6ilV5giwvbqUoOGbnXQnqZzUz/oL7OGZPYNoFKvy0=";
+    hash = "sha256-tgTH0W1KC1m6sye0o9blnnf5rAiwgnXSFTh6JSJh900=";
   };
 
-  nativeBuildInputs = [
-    llvmPackages.bintools
-    # gcc is used for the host tool, while clang is used for the bootloader.
-    llvmPackages.clang
-    llvmPackages.lld
-    mtools
-    nasm
-  ];
-
-  configureFlags = [
-    "--enable-all"
+  hardeningDisable = [
+    # clang doesn't support this for RISC-V target
+    "zerocallusedregs"
   ];
 
-  installFlags = [ "destdir=$out" "manprefix=/share" ];
+  nativeBuildInputs =
+    [
+      llvmPackages.libllvm
+      llvmPackages.lld
+    ]
+    ++ lib.optionals (enableAll || buildCDs) [
+      mtools
+    ]
+    ++ lib.optionals (hasI686 || hasX86_64) [ nasm ];
 
-  outputs = [ "out" "doc" "dev" "man" ];
+  outputs = [
```

The order is relevant. Now, `dev` is the default attribute.

```suggestion
  outputs = [
    out
```